src/Core/Traits/StatefulData.php

```diff
@@ -207,9 +207,13 @@ public function setState(array $data)
 			// Compare DateTime Objects
 			} elseif ($value instanceof \DateTimeInterface && $this->$key instanceof \DateTimeInterface) {
 				$current_key = $this->$key;
-				$interval = $value->diff($current_key);
 
-				if ($interval->format('F') > 0) {
+                $stored_date_ts = date_timestamp_get($current_key);
+                $received_date_ts = date_timestamp_get($value);
+                $timestamp_diff = abs($stored_date_ts - $received_date_ts);
```

Comment:
I was going to say you could use `$current_key->diff($value, true);` ... but I have no idea how you figure out from a DateInterval if there's actually a difference :(

Or according to the docs, you can just use comparison operators on dates
`$current_key == $value`. This makes me wonder why we were doing weird date formatting tricks to start with.

Reply:
Yeah, I was wondering this too. I think this might be a habit from Ye Olden Times when comparing date objects with `==` was discouraged.

I can't find where I saw the code in platform, but I think there was a place where date changes are ignored unless they're at least a day difference. Or I imagined it.